src/interp.c

```diff
@@ -445,7 +444,7 @@ void interpret(CHAR_DATA *ch, char *argument) {
     char command[MAX_INPUT_LENGTH];
     char logline[MAX_INPUT_LENGTH];
     const struct cmd_type *cmd;
-
+    memset(cmd_buf, 0, MAX_INPUT_LENGTH);
```

Comment:
Zeroing it here means we must have been accessing it in an uninitialised state somehow before? Like, relying on whatever was there before, which is still kinda nasty. If this is part of the fix (and I don't dispute it stops the crash) it suggests something elsewhere in interpret() is reading past where it should (and now into the "safe space" of zeros you prepare)